code/game/atoms.dm

```diff
@@ -336,6 +336,19 @@
 		return TRUE
 	return !density
 
+/**
+ * Return FALSE if nothing will prevent mover from entering the atom,
+ * whether it be the atom itself or an atom in its contents.
+ * Return TRUE if the atom could cross this atom while heading to the target atom
+ */
+/atom/proc/CouldEnter(atom/movable/mover, atom/next_loc)
```

Comment:
Yeah it's camel case, but since every similar proc have this i thought it was fitting

Reply:
The name of this proc is a bit unclear. This is a "can mover cross source towards next_loc but without actually Cross()'ing anything in source".